filters/tracing/statebagtotag.go

```diff
@@ -53,16 +53,22 @@ func NewStateBagToTag() filters.Spec {
 	return stateBagToTagSpec{}
 }
 
-func (f stateBagToTagFilter) Request(ctx filters.FilterContext) {
+func (f *stateBagToTagFilter) Request(ctx filters.FilterContext) {
+	value, ok := ctx.StateBag()[f.stateBagItemName]
+	if !ok {
+		return
+	}
+
 	span := opentracing.SpanFromContext(ctx.Request().Context())
 	if span == nil {
 		return
 	}
-	value, ok := ctx.StateBag()[f.stateBagItemName]
-	if !ok {
-		return
+
+	if _, ok := value.(string); ok {
+		span.SetTag(f.tagName, value)
+	} else {
+		span.SetTag(f.tagName, fmt.Sprint(value))
 	}
-	span.SetTag(f.tagName, fmt.Sprint(value))
 }
 
-func (stateBagToTagFilter) Response(ctx filters.FilterContext) {}
+func (f *stateBagToTagFilter) Response(ctx filters.FilterContext) {}
```

Comment:
You don't need to assign it to a name, or any linter that yells?

Reply:
I remember there was a linter warning in the past so I changed that but maybe its not needed anymore

Reply:
Removed the name